### PR TITLE
chore: update git create organisation resolver to support for AWS code commit

### PR DESCRIPTION
**Pull request by overbit**
Close: #6351 

### Summary
🚀🛠️🐛

Added support for AWS CodeCommit as a git provider option and for generating GraphQL types and hooks for the client package. Refactored the models and inputs for creating git organizations to support multiple providers. Updated the GraphQL schema and the client and server code to use the new models and inputs. Added and modified dependencies and tasks for the GraphQL codegen. Sorted the GraphQL schema types alphabetically.

> _We're adding more git providers to our app_
> _So we can push our code to any repo_
> _We use the codegen to make our hooks and types_
> _And we validate our props with a map_

### Walkthrough
*  Add GraphQL codegen plugins and documents for generating TypeScript types and React hooks for GraphQL operations in the client package ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-e0a5e4afad70170535bbca42b530798554a4f85049cd10e16c6903b281eb1e56L9-R19), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R192-R193), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-aa01f023fa5f97523fa4dda50e7002cb125f8a88bf2a12fbf2deec2b8d8fc26dR20), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-b9ccc53c48e521fb7c98b07ad7f4761ba0ea91eb649e157d159b065ac2487c1dR1-R2), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-b9ccc53c48e521fb7c98b07ad7f4761ba0ea91eb649e157d159b065ac2487c1dR10), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-b9ccc53c48e521fb7c98b07ad7f4761ba0ea91eb649e157d159b065ac2487c1dL2177-R2338), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-fcc07755705d42d75e63cb9d38f7d97fe2e46782537b5cda2934816c1ae65486R1-R57))
*  Update `createOrganization` mutation to support multiple git providers with different inputs ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-8e34f7703a0b39fa5e612bb39ad750167d90a6b75df578f54f35a9348dd0a509L801-R819), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-8e34f7703a0b39fa5e612bb39ad750167d90a6b75df578f54f35a9348dd0a509L892), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-b9ccc53c48e521fb7c98b07ad7f4761ba0ea91eb649e157d159b065ac2487c1dL801-R822), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-b9ccc53c48e521fb7c98b07ad7f4761ba0ea91eb649e157d159b065ac2487c1dL892), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-3860f234d2c5629bf7aeb4e7439fe3b357be4e00c08a9f008b56d3610bb017c0L801-R819), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-3860f234d2c5629bf7aeb4e7439fe3b357be4e00c08a9f008b56d3610bb017c0L892), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-6137baa91c68c17d5e3716428198a16a3cb5cdd263e22998503daa29c86a296eL801-R819), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-6137baa91c68c17d5e3716428198a16a3cb5cdd263e22998503daa29c86a296eL892), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-11ded4f4d03080d0b3917fc193a7aa873567794357733d269271b1075f25d7a0L1-R17), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-11ded4f4d03080d0b3917fc193a7aa873567794357733d269271b1075f25d7a0L8-R32))
*  Use custom hooks generated by GraphQL codegen instead of `useMutation` hook from `@apollo/client` in callback components ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-6a5da699abe2fad9901dd52129afd957e480d92ca77fe49cb67c581335f2453eL2-R13), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-255e9d9d9cb2232f05646c62b005b95dbbe0b94c4ba46ca22747a83ee682878dL2-R13))
*  Add AWS CodeCommit feature and logo to the client package ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-3da58499a47d508bd6bff003ed7125c5408d48b6b708e3d5b1df63aa0ee755b0L4-R9), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-3da58499a47d508bd6bff003ed7125c5408d48b6b708e3d5b1df63aa0ee755b0R15), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-5af106441ba9187b6775a40d93abe24a8d709f56acd1bb7a60d58a1167fe4ef8R43-R45), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-5af106441ba9187b6775a40d93abe24a8d709f56acd1bb7a60d58a1167fe4ef8R93-R97), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-f2d4a72a4f2ee7f10179ea93ccc95c5a8129e6a99206d3e263fa920b3b6a4282R10))
*  Refactor `GitFactory` class to use a map of validation functions for git provider organization properties ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7L3-R3), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7R9), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7L26-R22), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7L39-R35), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-566383ea73f182bbd64b7db276dd9741a837673ce886790103fe63d0dcee6ad7L50-R48), [link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-c7847e74f0348024779e0a630ffefa1d22d06f3723c79ef0a213d7cc55d7da0cR1-R13))
*  Change `init` method of `AwsCodeCommitService` class to return an empty promise instead of throwing an error ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-79b256dbe34679386e43a1af887c0d3a77531fcf3f7b69683a0dffe258c7262aL102-R102))
*  Enable sorting of GraphQL schema types alphabetically ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-4e72dc01e875c6474e7255d4778cb9acef369010c0ae3f813255eeee5fabcc24R43))
*  Add exclamation mark to `data` argument of `createProject` mutation ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-53fe876ec63d5bae3535375cdebaa519819e0458bd8bade47aa42f9a3f1e1dafL43-R43))
*  Delete unused file `git-callback.ts` ([link](https://github.com/amplication/amplication/pull/6436/files?diff=unified&w=0#diff-d7a6c35534002b67310ce11a7b2a30e5849121168cfce9288a269aca122e816e))

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Resource/git/AuthResourceWithBitbucketCallback.tsx [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+  EnumGitProvider,
+  useCompleteGitOAuth2FlowMutation,
+} from "../../models";
 import { GET_PROJECTS } from "../../Workspaces/queries/projectQueries";
 
 const AuthResourceWithBitbucketCallback = () => {
   const { trackEvent } = useTracking();
-  const [completeAuthWithGit] = useMutation<boolean>(COMPLETE_OAUTH2_FLOW, {
+
+  const [completeAuthWithGit] = useCompleteGitOAuth2FlowMutation({
```
Comment (abrl91) [inline packages/amplication-client/src/Resource/git/AuthResourceWithBitbucketCallback.tsx:13]: Is this related to what you did with graphql codegen?

File: packages/amplication-client/src/Resource/git/queries/git-callback.query.ts
Comment (abrl91) [inline packages/amplication-client/src/Resource/git/queries/git-callback.query.ts:1]: nice!

File: packages/amplication-client/src/Resource/git/GitActions/GitProviderConnectionList.tsx [hunk 2]
```diff
@@ -87,6 +90,11 @@ export const GitProviderConnectionList: React.FC<Props> = ({
         onSyncNewGitOrganizationClick={handleAddProvider}
         disabled={!showBitbucketConnect.hasAccess}
       />
+      <GitProviderConnection
+        provider={EnumGitProvider.AwsCodeCommit}
+        onSyncNewGitOrganizationClick={handleAddProvider}
```
Comment (abrl91) [inline packages/amplication-client/src/Resource/git/GitActions/GitProviderConnectionList.tsx:95]: Does it work? I don't see that you changed or added something related to this

File: packages/amplication-client/src/Workspaces/queries/projectQueries.ts [hunk 3]
```diff
@@ -40,7 +40,7 @@ export const GET_PROJECTS = gql`
 `;
 
 export const CREATE_PROJECT = gql`
-  mutation createProject($data: ProjectCreateInput) {
+  mutation createProject($data: ProjectCreateInput!) {
```
Comment (abrl91) [inline packages/amplication-client/src/Workspaces/queries/projectQueries.ts:43]: 💪

File: packages/amplication-client/src/util/BillingFeature.ts [hunk 4]
```diff
@@ -7,6 +7,7 @@ export enum BillingFeature {
   CodePushToGit = "feature-code-push-to-git",
   IgnoreValidationCodeGeneration = "feature-ignore-validation-code-generation",
   HideNotifications = "feature-ignore-validation-code-generation-hide-notifications",
+  AwsCodeCommit = "feature-awscodecommit",
```
Comment (abrl91) [inline packages/amplication-client/src/util/BillingFeature.ts:10]: Did you add it only to the client? I think we should also add it to the server and add a check for the entitlement on the entry of the function, like we did with Bitbucket and import DB

File: packages/amplication-server/src/core/git/dto/inputs/GitOrganizationCreateInput.ts [hunk 5]
```diff
@@ -1,17 +1,33 @@
...
+
+// // eslint-disable-next-line @typescript-eslint/naming-convention
+// export const GitOrganizationCreateInput = createUnionType({
+//   types: () =>
+//     [
+//       GitOrganizationCreateInputAwsCodeCommit,
+//       GitOrganizationCreateInputGitHub,
+//     ] as const,
+// });
+
```
Comment (abrl91) [inline packages/amplication-server/src/core/git/dto/inputs/GitOrganizationCreateInput.ts:13]: can we delete it?

File: packages/amplication-client/src/Resource/git/AuthResourceWithBitbucketCallback.tsx (hunk 1 above)
Comment (overbit) [reply packages/amplication-client/src/Resource/git/AuthResourceWithBitbucketCallback.tsx:13]: yep, a pure 1-2-1 replacement

File: packages/amplication-client/src/Resource/git/GitActions/GitProviderConnectionList.tsx (hunk 2 above)
Comment (overbit) [reply packages/amplication-client/src/Resource/git/GitActions/GitProviderConnectionList.tsx:95]: let me remove this for now so it would be just a placeholder to start with

File: packages/amplication-client/src/util/BillingFeature.ts (hunk 4 above)
Comment (overbit) [reply packages/amplication-client/src/util/BillingFeature.ts:10]: Added to the server too. Good catch

**Review by abrl91 — APPROVED** [review]
